src/dparse/parser.d

```diff
@@ -3560,60 +3560,82 @@ class Parser
     IfStatement parseIfStatement()
     {
         mixin(traceEnterAndExit!(__FUNCTION__));
-        auto node = allocator.make!IfStatement;
+        IfStatement node = allocator.make!IfStatement;
         node.line = current().line;
         node.column = current().column;
         mixin(tokenCheck!"if");
         if (moreTokens)
             node.startIndex = current().index;
         mixin(tokenCheck!"(");
+        Bookmark b = setBookmark();
 
-        if (currentIs(tok!"auto"))
+        // most commn case: rel expression and such
+        if (Expression e = parseExpression())
+            if (moreTokens && currentIs(tok!")"))
```

Comment:
The need for this reveals a bug:

    if (int i = a < 100) b(); 

An expression is found before reaching the right paren.